lib/src/directory_watcher/windows.dart

```diff
@@ -184,7 +184,7 @@ class _WindowsDirectoryWatcher
           if (_files.containsDir(path)) continue;
 
           var stream = Directory(path).list(recursive: true);
-          StreamSubscription<FileSystemEntity> subscription;
+          late StreamSubscription<FileSystemEntity> subscription;
```

Comment:
We should be able to avoid the `late` here.

Can you split up the `onDone` and `onError` handlers to be assigned after the subscription is created?

```dart
var subscription = stream.listen((entity) {
    //...
}, cancelOnError: true);
subscription.onDone(() {
    _listSubscriptions.remove(subscription);
});
subscription.onError((Object error, StackTrace stackTrace) {
    _listSubscriptions.remove(subscription);
    _emitError(e, stackTrace);
});
```

Reply:
Done, here and in `mac_os.dart`.